base/src/com/google/idea/blaze/base/projectview/section/sections/TextBlock.java

```diff
@@ -28,30 +30,37 @@
 public class TextBlock implements Serializable {
   private static final long serialVersionUID = 1L;
   private final ImmutableList<String> lines;
+  public final int firstLineIndex;
 
-  public TextBlock(ImmutableList<String> lines) {
+  public TextBlock(ImmutableList<String> lines, int firstLineIndex) {
     this.lines = lines;
+    this.firstLineIndex = firstLineIndex;
   }
 
   /** Returns raw lines, including any indentation and surrounding whitespace */
   public ImmutableList<String> lines() {
     return lines;
   }
 
-  public static TextBlock of(String... lines) {
-    return new TextBlock(ImmutableList.<String>builder().add(lines).build());
+  public static TextBlock of(int firstLineIndex, String... lines) {
+    return new TextBlock(ImmutableList.<String>builder().add(lines).build(), firstLineIndex);
   }
 
   /** A text block that is a single newline */
-  public static TextBlock newLine() {
-    return new TextBlock(ImmutableList.of(""));
+  public static TextBlock newLine(int firstLineIndex) {
```

Comment:
If this is always going to be -1 at creation time (I don't see a line in this PR where we pass a different value), could we just not pass it and assign -1 to it?

Reply:
We could, but I can totally imagine a future implementation where we would add a new line at an existing file. In that case, we would know the line number AND we would wnt to specify where to do that exactly. One example would be: a linter/formatter.